dv/uvm/core_ibex/sim.py

```diff
@@ -1,301 +1,512 @@
+            os.makedirs(dirname, exist_ok=True)
+            cmds.append(cmd)
+        run_parallel_cmd(cmds, 600, check_return_code=check_return_code)
+        return
+
+    # We're not in LSF mode, so we'll create the output directories as we go.
+    # That should make it a bit easier to see how far we got if there was an
+    # error.
+    for desc, cmd, dirname in command_list:
+        os.makedirs(dirname, exist_ok=True)
+        logging.info("Running " + desc)
+        run_cmd(cmd, 300, check_return_code=check_return_code)
+
+
+def rtl_sim(sim_cmd, test_list, seed, opts,
+            output_dir, bin_dir, lsf_cmd, check_return_code):
+    """Run the testbench in the simulator
+
+    sim_cmd is the base command (as returned by get_simulator_cmd). This will
+    still have placeholders for test-specific arguments. test_list is a list of
+    test objects read from the testlist YAML file which gives the tests to run.
+
+    seed is the seed to use in the simulations (controls things like random
+    delays on the bus). opts is a string of plusargs to give to the simulator.
+
+    output_dir is the output directory for simulation files (and the directory
+    in which the simulator gets run). bin_dir is the directory containing
+    binaries to be run.
+
+    If lsf_cmd is not None, it should be prefixed on each command, which will
+    be run in parallel.
+
+    check_return_code is True if we should check the return codes from
+    simulator executions.
+
+    """
+    logging.info("Running RTL simulation...")
+
+    sim_cmd = subst_vars(sim_cmd,
+                         {
+                             'out': output_dir,
+                             'sim_opts': opts,
+                             'cwd': _CORE_IBEX,
+                             'seed': str(seed)
+                         })
+
+    # Compute a list of pairs (cmd, dirname) where cmd is the command to run
+    # and dirname is the directory in which the command should be run.
+    cmd_list = []
+    for test in test_list:
+        for i in range(test['iterations']):
+            cmd_list.append(get_test_sim_cmd(sim_cmd, test, i,
+                                             output_dir, bin_dir, lsf_cmd))
+
+    run_sim_commands(cmd_list, lsf_cmd is not None, check_return_code)
+
+
+def compare_test_run(test, idx, iss, output_dir, report):
+    '''Compare results for a single run of a single test
+
+    Here, test is a dictionary describing the test (read from the testlist YAML
+    file). idx is the iteration index. iss is the chosen instruction set
+    simulator (currently supported: spike and ovpsim). output_dir is the base
+    output directory (which should contain logs from both the ISS and the test
+    run itself). report is the path to the regression report file we're
+    writing.
+
+    Returns True if the test run passed and False otherwise.
+
+    '''
+    test_name = test['test']
+    elf = os.path.join(output_dir, '{}.{}.o'.format(test_name, idx))
```

Comment:
I think this should be `os.path.join(output_dir, 'instr_gen/asm_tests/{}.{}.o'.format(test_name, idx))`

Reply:
Yes, you're completely right. Will fix.